.github/scripts/check-anchors.py

```
@@ -0,0 +1,248 @@
+    ap.add_argument('--file', action='append', default=[],
+                    help='Check a specific file (repeatable)')
+    args = ap.parse_args()
+
+    root = os.path.abspath(args.root)
+
+    if args.file:
+        files = [os.path.abspath(f) for f in args.file]
+    elif args.only_changed:
+        # PR: compare against origin/main if available; fall back to HEAD~1
+        # for direct pushes to main (where origin/main == HEAD).
+        out = subprocess.run(
```

Ruff (0.16.1)

[warning] `subprocess.run` without explicit `check` argument

Add explicit `check=False`

(PLW1510)